### ISSUE TITLE
Filter entity selector by integration doesn't work

### ISSUE THREAD

**Issue by balloob** [home-assistant/frontend#12254]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

We got reports that using the utility meter service for resetting didn't allow selecting the sensors. This means that filtering an entity list by integration doesn't work. Selector config:

![image](https://user-images.githubusercontent.com/1444314/162326013-4f304104-119a-4cb2-9ffa-2d095a43cfef.png)

(the red line is because it's getting removed for this HA release)

### Describe the behavior you expected

It should show only the entities created by the utility meter.

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

core-2022.4

### What was the last working version of Home Assistant Core?

-

**Comment by frenck** [comment]
possibly related to #12246

**Comment by spacegaier** [comment]
My PR should not be related and in fact it works just fine for me (in dev with and without my PR).

**Comment by balloob** [comment]
CC @dgomes 

I wonder if we should add diagnostics to utility so users can share their config

**Comment by dgomes** [comment]
Yes, I'm planning to add diagnostics to `utility_meter` and `integration` as it now became very difficult to collect users config during debug...

One question, is it possible to have diagnostics dump an image (even if b64 encoded in the log file) ? It's quite useful to have the graph of the source sensor...

**Comment by dgomes** [comment]
Just realised `utility_meter` and `integration` are not `devices`, how should the user trigger diagnostics ?
